### PR TITLE
Queues - get approximate message count (fixes #199)

### LINKED ISSUE #199: Queues - get approximate message count
per the rest api: https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx

### PR DESCRIPTION
**Pull request by kenegozi**

### CODE REVIEW

File: storage/queue.go [hunk 1]
```diff
@@ -111,6 +111,44 @@ type PeekMessageResponse struct {
 	MessageText    string `xml:"MessageText"`
 }
 
+// QueueMetadata represents queue properties on
+// a specific queue.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx
+type QueueMetadata struct {
```
Comment (ahmetalpbalkan) [inline storage/queue.go]: Please call this `QueueMetadataResponse`

File: storage/queue.go [hunk 2]
```diff
@@ -111,6 +111,44 @@ type PeekMessageResponse struct {
...
+	headers := c.client.getStandardHeaders()
+	resp, err := c.client.exec("GET", uri, headers, nil)
+	if err != nil {
+		return qm, err
+	}
+	defer resp.body.Close()
+
+	for k, v := range resp.headers {
+		if len(v) != 1 {
+			return qm, fmt.Errorf("Unexpected number of valuus (%d) in response header '%s'", len(v), k)
```
Comment (ahmetalpbalkan) [inline storage/queue.go]: typo valuus

File: storage/queue.go [hunk 3]
```diff
@@ -111,6 +111,44 @@ type PeekMessageResponse struct {
...
+		}
+
+		if k == approximateMessagesCountHeader {
```
Comment (ahmetalpbalkan) [inline storage/queue.go]: how do you know the uppercase/lowercase will match? Server may send lowercase headers or `X-ms-approximate-messages-count` which is ≠ `X-Ms-Approximate-Messages-Count`.

File: storage/queue.go [hunk 4]
```diff
@@ -228,3 +266,6 @@ func (c QueueServiceClient) DeleteMessage(queue, messageID, popReceipt string) e
 	defer resp.body.Close()
 	return checkRespCode(resp.statusCode, []int{http.StatusNoContent})
 }
+
+const approximateMessagesCountHeader = "X-Ms-Approximate-Messages-Count"
+const userDefinedMetadataHeaderPrefix = "X-Ms-Meta-"
```
Comment (ahmetalpbalkan) [inline storage/queue.go]: unused

File: storage/queue.go [hunk 5]
```diff
@@ -228,3 +266,6 @@ func (c QueueServiceClient) DeleteMessage(queue, messageID, popReceipt string) e
 	defer resp.body.Close()
 	return checkRespCode(resp.statusCode, []int{http.StatusNoContent})
 }
+
+const approximateMessagesCountHeader = "X-Ms-Approximate-Messages-Count"
```
Comment (ahmetalpbalkan) [inline storage/queue.go]: please to the beginning of the file after the imports.

File: storage/queue.go (hunk 3 above)
Comment (kenegozi) [inline storage/queue.go]: Golang's http client is canonicalizing the http.Header collection this way.

btw I did run the enclosed tests against a live azure storage account and they pass, so it definitely is working.

Comment (ahmetalpbalkan) [inline storage/queue.go]: I see, thanks for confirming.

**Comment by kenegozi** [comment]
this change is now included in  #202 with the comments from here taken care of (fixing the consts, valuus typo, response type name)

**Pull request closed without merging**